src/mixer.cpp

```diff
@@ -188,24 +193,32 @@ void *mixer_thread(void *param) {
 						channel->axcindicate = SIGNAL;
 					}
 					input->ready = false;
-					RESET_BIT(mixer->inputs_todo, j);
+					mixer->inputs_todo[j] = 0;
 				}
 				pthread_mutex_unlock(&input->mutex);
 			}
 
-			if((mixer->inputs_todo & mixer->input_mask) == 0 || mixer->interval == 0) {	// all good inputs handled or last interval passed
+			unprocessed_mixer_count = 0;
+			for(int k = 0; k < MAX_MIXINPUTS; k++) {
+				unprocessed_mixer_count = unprocessed_mixer_count + ( mixer->inputs_todo[k] & mixer->input_mask[k]);
+            }
+			if((unprocessed_mixer_count == 0) || mixer->interval == 0) {	// all good inputs handled or last interval passed
 #ifdef DEBUG
 				gettimeofday(&te, NULL);
-		        debug_bulk_print("mixerinput: %lu.%lu %lu int=%d inp_unhandled=0x%02x inp_mask=0x%02x\n",
-					te.tv_sec, (unsigned long) te.tv_usec, (te.tv_sec - ts.tv_sec) * 1000000UL + te.tv_usec - ts.tv_usec,
-					mixer->interval, mixer->inputs_todo, mixer->input_mask);
+                for(int k = 0; k < mixer->input_count; k++) {
```

Comment:
this seems less than ideal.  these debug printouts are already quite verbose, adding another line for each mixer input with the only difference being `mixer->inputs_todo[k]`, `mixer->input_mask[k]` is a lot more output.  How about generating a single string that has the list of true / false for each, and including that once?